tests/integration/integration_nodejs_test.go

```diff
@@ -2410,11 +2410,17 @@ func TestPackageAddProviderFromRemoteSource(t *testing.T) {
 	e.Env = append(e.Env, "PULUMI_DISABLE_AUTOMATIC_PLUGIN_ACQUISITION=false")
 	e.RunCommand("pulumi", "stack", "select", "organization/packageadd-remote", "--create")
 
-	// The provider currently lives in the test-plugin branch in pulumi/pulumi.  Once we allow
-	// plugins installed from subdirectories, this should move into a subdirectory of tests/integration
-	// and the test-plugin branch can be deleted.
+	// The broken provider doesn't succeed in generating the SDK.  We still want to see it installed,
+	// and want to make sure we can still install a different provider from a different subdirectory,
+	// from the same repository and the same revision.
+	e.RunCommandExpectError("pulumi", "package", "add",
+		"github.com/pulumi/component-test-providers/broken-test-provider@d47cf0910e0450400775594609ee82566d1fb355")
+	stdout, _ := e.RunCommand("pulumi", "plugin", "ls")
+	require.Contains(t, stdout, "github.com_pulumi_component-test-providers")
+	require.Contains(t, stdout, "0.0.0-xd47cf0910e0450400775594609ee82566d1fb355")
+
 	e.RunCommand("pulumi", "package", "add",
-		"github.com/pulumi/pulumi@178af2fc9a5da4b839e53681fadc4a0771b866ea")
+		"github.com/pulumi/component-test-providers/test-provider@d47cf0910e0450400775594609ee82566d1fb355")
```

Comment:
Could you add a README to https://github.com/pulumi/component-test-providers explaining that they are used in these tests? I don't want anyone deleting/archiving the repo by mistake :D

Reply:
Yep, good idea, done!

Reply:
Why couldn't we keep using pulumi/pulumi?

Reply:
Two reasons:
- `pulumi/pulumi` is big, and cloning it every time in an integration test is slow.  I'd rather avoid this.
- The current implementation of tarballs doesn't support symlinks.  `pulumi/pulumi` contains some of them, so it doesn't work right now.  This is going to be fixed in https://github.com/golang/go/issues/49580#issuecomment-2518721262, which according to that comment should land in Go 1.25. We could hand-roll more code if we decide it's worth it until that lands, but it's more complicated, and didn't seem worth it for now. Happy to discuss this part though.